File: Gevorgyan.TodoListApp.Application/Commands/Users/Management/Models/CreateUserInputModel.cs
using System.ComponentModel.DataAnnotations;

namespace Gevorgyan.TodoListApp.Application.Commands.Users.Management.Models
{
    /// <summary>
    /// Input model for creating user
    /// </summary>
    public class CreateUserInputModel : InputModelBase
    {
        /// <summary>
        /// E-mail
        /// </summary>
        [RequiredInModel]
        [Display(Name = "Email of user")]
        public string Email { get; set; }

        /// <summary>
        /// Name
        /// </summary>
        [RequiredInModel]
        [Display(Name = "Name of user")]
        public string Name { get; set; }
        
        /// <summary>
        /// Password
        /// </summary>
        [RequiredInModel]
        [NotTrimable]
        [Display(Name = "Password of user")]
        public string Password { get; set; }

        /// <summary>
        /// Password confirmation 
        /// </summary>
        [RequiredInModel]
        [NotTrimable]
        [Display(Name = "Password confirmation of user")]
        public string PasswordConfirmation { get; set; }
    }
}

File: Gevorgyan.TodoListApp.RestApi/ClientApp/src/app/core/models/CreateUserInputModel.ts



// for AUTO-GENERATION!



/*
 * Input model for creating user
 */
export interface ICreateUserInputModel {
    
    /*
     * E-mail
     */
    email: string;

    /*
     * Name
     */
    name: string;

    /*
     * Password
     */
    password: string;

    /*
     * Password confirmation
     */
    passwordConfirmation: string;
}
    
/*
 * Input model for creating user
 */
export class CreateUserInputModel implements ICreateUserInputModel {
    
    /*
     * E-mail
     */
    email: string;
    
    /*
     * Name
     */
    name: string;
    
    /*
     * Password
     */
    password: string;
    
    /*
     * Password confirmation
     */
    passwordConfirmation: string;
    
    constructor(data?: ICreateUserInputModel) {
        
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            
            this.email = data["email"];
            this.name = data["name"];
            this.password = data["password"];
            this.passwordConfirmation = data["passwordConfirmation"]; 
        }
    }

    static fromJS (data: any): CreateUserInputModel {
        data = typeof data === 'object' ? data : {};
        let result = new CreateUserInputModel;
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        
        data["email"] = this.email;
        data["name"] = this.name;
        data["password"] = this.password;
        data["passwordConfirmation"] = this.passwordConfirmation;
        
        return data; 
    }
}    
